src/excmds.ts

```diff
@@ -463,3 +477,4 @@ async function listTabs() {
 // }}}
 
 // }}}
+// vim: tabstop=4 shiftwidth=4 expandtab
```

Comment:
Maybe add `foldmethod=marker`? I don't know how much other people like folds.

Reply:
I never use folds FWIW.

Reply:
I'm using folds a lot in my stuff ;-)
I'll add the stuff you mentioned in review later.

Reply:
Are modelines usually in the footer? I never use them myself. I know [Mozilla coding style][0] is to put it in the beginning, but I'm aware we are not Mozilla. I think they are more visible at the top so anyone not too keen on having their environment changed can see them right away (and if we want to start enforcing style, a linter would be more effective, I feel).

I also rarely use folds coding. Wikis, however...

[0]: https://developer.mozilla.org/en-US/docs/Mozilla/Developer_guide/Coding_Style#Mode_Line

Reply:
I've only ever seen them at EOF (I actually assumed they had to be there).

The plan is to use a format tool such as prettier, but we need to fix the macro script to deal with multi line function declarations first.

Reply:
FWIW, I use vim-sleuth to automatically choose tabstop, shiftwidth, etc.

Reply:
It's a styling thing but having them on the beginning is good because they're never moved.
On the other hand having them at the end is less disrupting for non-vim users. And IMHO we don't need emacs modeline as clearly nobody here uses emacs (not in a vim-like project) \m/

Reply:
Right, you can do `set nomodeline` so anyone really bothered should be using that. After that, I think we should add this modeline to each file for consistency. I don't feel one extra line at the top is significantly more distracting than one at the bottom, but I'm not too invested in this either way.

Reply:
I'll do that later.